frontend/src/libs/import-export/normalize-ddah-import.ts

```diff
@@ -0,0 +1,111 @@
+import { Applicant, MinimalDdah } from "../../api/defs/types";
+import {
+    SpreadsheetRowMapper,
+    matchByUtoridOrName,
+} from "../../libs/import-export";
+
+/**
+ * Convert imported spreadsheet or JSON data into an
+ * array of minimal DDAH objects.
+ *
+ * @param {({
+ *     fileType: "json" | "spreadsheet";
+ *     data: any;
+ * })} data
+ * @param {Applicant[]} applicants
+ * @returns {MinimalDdah[]}
+ */
+export function normalizeDdahImports(
+    data: {
+        fileType: "json" | "spreadsheet";
+        data: any;
+    },
+    applicants: Applicant[]
+): MinimalDdah[] {
+    let ret: MinimalDdah[] = [];
+
+    if (data.fileType === "json") {
+        let unwrapped: MinimalDdah[] = data.data;
+        if ((unwrapped as any).ddahs) {
+            unwrapped = (unwrapped as any).ddahs;
+        }
+        for (const ddah of unwrapped) {
+            ret.push(ddah);
+        }
+    }
+
+    if (data.fileType === "spreadsheet") {
+        const unwrapped = data.data;
+        // Get an upper bound for the maximum number of duties that the spreadsheet might have
+        const maxDuties = Math.round(
+            Math.max(
+                ...unwrapped.map((row: object) => Object.keys(row).length),
+                0
+            ) / 2
+        );
+
+        // We need to generate a keymap for all the likely column names
+        const keyMap: { [key: string]: string } = {
+            Position: "position_code",
+            "First Name": "first_name",
+            "Given Name": "first_name",
+            First: "first_name",
+            "Last Name": "last_name",
+            Surname: "last_name",
+            "Family Name": "last_name",
+            Last: "last_name",
+        };
+        // We will also add `Hours #` and `Duty #` to the keymap for the number of duties in our range
+        for (let i = 0; i <= maxDuties; i++) {
+            keyMap[`Duty ${i}`] = `duty_${i}`;
```

Comment:
Is this part correct? So duties with numbers less than 10 will have 2 different entries? For example duty 1 will have 2 keys 'Duty 1' and 'Duty 01' with same value 'duty_1'.

Reply:
This is correct. The purpose is to recognize both 'Duty 1' and 'Duty 01' both as a valid key-mappings to 'duty_1'.